main.go

```diff
@@ -49,6 +60,33 @@ env vars: GPTSCRIPT_WORKSPACE_DIR`)
 	}
 }
 
+func handleGzip(elements string) (string, error) {
+	var gz struct {
+		Content string `json:"_gz"`
+	}
+	if err := json.Unmarshal([]byte(elements), &gz); err != nil {
+		// If it didn't unmarshal, then it's not gzipped
+		return elements, nil
+	}
+
+	decoded, err := base64.StdEncoding.DecodeString(gz.Content)
+	if err != nil {
+		return "", err
+	}
+
+	reader, err := gzip.NewReader(bytes.NewReader(decoded))
+	if err != nil {
+		return "", err
+	}
```

Comment:
Need to close the reader here

```suggestion
	reader, err := gzip.NewReader(bytes.NewReader(decoded))
	if err != nil {
		return "", err
	}
	defer reader.close()
```